### ISSUE TITLE
[Bug]: The error message returned by creating the BIN_IVF_FLAT index with L2 metric type is wrong

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#26038]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.0-20230731-26d1c9cd
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.2.15.dev3
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. create index `BIN_IVF_FLAT` with metric_type L2 and return the error: `supported: [HAMMING JACCARD TANIMOTO SUBSTRUCTURE SUPERSTRUCTURE]`. This error message is misleading, index `BIN_IVF_FLAT` only support metrics type: [HAMMING JACCARD TANIMOTO]
![image](https://github.com/milvus-io/milvus/assets/27288593/c5b87b81-6196-4e1b-a0ce-b32660ddb361)

```
[2023-07-31 19:54:51 - DEBUG - ci_test]: (api_request)  : [Collection.create_index] args: ['binary_vector', {'index_type': 'BIN_IVF_FLAT', 'metric_type': 'L2', 'params': {'nlist': 64}}], kwargs: {'timeout': 1200, 'index_name': 'binary_vector'} (api_request.py:56)
[2023-07-31 19:54:51 - ERROR - pymilvus.decorators]: RPC error: [create_index], <MilvusException: (code=1, message=metric type not found or not supported, supported: [HAMMING JACCARD TANIMOTO SUBSTRUCTURE SUPERSTRUCTURE])>, <Time:{'RPC start': '2023-07-31 19:54:51.124351', 'RPC error': '2023-07-31 19:54:51.143142'}> (decorators.py:126)
[2023-07-31 19:54:51 - ERROR - ci_test]: Traceback (most recent call last):
  File "/home/zong/zong/projects/milvus/tests/python_client/utils/api_request.py", line 26, in inner_wrapper
    res = func(*args, **_kwargs)
  File "/home/zong/zong/projects/milvus/tests/python_client/utils/api_request.py", line 57, in api_request
    return func(*arg, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 1082, in create_index
    return conn.create_index(self._name, field_name, index_params, timeout=timeout, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 127, in handler
    raise e from e
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 123, in handler
    return func(*args, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 162, in handler
    return func(self, *args, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 102, in handler
    raise e from e
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 68, in handler
    return func(*args, **kwargs)
  File "/home/zong/zong/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 770, in create_index
    raise MilvusException(status.error_code, status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=metric type not found or not supported, supported: [HAMMING JACCARD TANIMOTO SUBSTRUCTURE SUPERSTRUCTURE])>
 (api_request.py:39)
[2023-07-31 19:54:51 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=metric type not found or not supported, supported: [HAMMING JACCARD TANIMOTO SUBSTRUCTURE SUPERSTRUCTURE])> (api_request.py:40)
```
Please update the error message like metric type `SUBSTRUCTURE`
create index `BIN_IVF_FLAT` with metric_type and return the error: `  <MilvusException: (code=1, message=metric type not found or not supported, supported: [HAMMING JACCARD TANIMOTO])> `

### Expected Behavior

Update error image to ` <MilvusException: (code=1, message=metric type not found or not supported, supported: [HAMMING JACCARD TANIMOTO])>`

### Steps To Reproduce

```markdown
def test_tmp_index(self):
        c_name = cf.gen_unique_str(prefix)
        collection_w = self.init_collection_wrap(name=c_name, schema=default_binary_schema)

        df, _ = cf.gen_default_binary_dataframe_data()
        ins_res, _ = collection_w.insert(df)

        binary_index_params = {"index_type": "BIN_IVF_FLAT", "metric_type": "L2", "params": {"nlist": 64}}
        collection_w.create_index(default_binary_vec_field_name, binary_index_params,
                                  index_name=binary_field_name)
        assert collection_w.has_index(index_name=binary_field_name)[0] == True
        assert len(collection_w.indexes) == 1
        log.info(collection_w.indexes[0].params)
```
```


### Milvus Log

_No response_

### Anything else?

_No response_

**Comment by smellthemoon** [comment]
pr merged, plz help verify. @ThreadDao

**Comment by ThreadDao** [comment]
Fixed 2.2.0-20230808-506ec414
